index.html

```diff
@@ -819,20 +819,26 @@
 		</section>
 		</div>
 		<section>
+			<h2 id="tracking-legacy-mouse-pos">Tracking the effective position of legacy mouse pointer</h2>
+			<div>
+			<p>While only <a href="#the-primary-pointer">primary pointers</a> can produce compatibility mouse events, <a href="#multiple-primary-pointers">multiple primary pointers</a> can be active simultaneously, each producing its own stream of compatibility mouse events.  Since all these compatibility events would appear to a legacy mouse-specific code to be coming from a single mouse hardware, user agents are encouraged to guarantee that the compatibility mouse events are consistent from a single device perspective.  For mouse transition events (i.e., <code>mouseover</code>, <code>mouseout</code>, <code>mouseenter</code> and <code>mouseleave</code>), this means the entry/exit state for every event target is valid as implied by [<cite><a class="bibref" href="#bib-DOM-LEVEL-3-EVENTS">DOM-LEVEL-3-EVENTS</a></cite>].  Users agents MAY guarantee this by maintaining the <dfn id="dfn-legacy-mouse-pos">effective position of legacy mouse pointer</dfn> in the document as follows.</p>
+			<p>Right before firing a <code>pointerdown</code>, <code>pointerup</code> or <code>pointermove</code> event, the user agent MAY run the following steps:</p>
+			<ol>
+				<li>Let <code>T</code> be the target of the event above.</li>
+				<li>If <code>T</code> is equal to current <a href="#dfn-legacy-mouse-pos" class="internalDFN">effective legacy mouse pointer position</a>, terminate these steps.</li>
```

Comment:
The way I see it there are two styles of spec writing - algorithm heavy and prose heavy.  The WhatWG DOM spec is algorithm heavy - very precisely defines what is supposed to happen but can be harder to follow.  The UI Events spec is prose heavy - you get the basic idea pretty easily but there can be ambiguity in the edge cases (though if that's resolved by precise tests then it should be fine IMHO).  I like that you're trying to define an algorithm here, but it's hard to do well when you're referring to stuff that's defined in a more handy-wavy prose style elsewhere.

So if we want to be consistent with the UI Event spec treatment, we could just remove the algorithm and definition and say (somewhat vaguely) that the transition events follow from the UI Events definition corresponding to a single mouse being moved around.

Or if you'd like to try to make it more precise, then yes I think you need some text around the default value, and unsetting it on pointerleave from the window etc.  That's a really going point actually - as written now it looks like it's illegal to send a mouseleave event when the mouse leaves the window (because there is no corresponding `pointermove` sent).  If you want to keep the algorithm style you'll need to expand the usage of this section to handle that special case.  Note that specs usually use the simpler concept of "is set" and "unset" rather than using a programming value like "NULL" (these aren't real variables afterall).